src/main/scala/io/iohk/ethereum/db/dataSource/IodbDataSource.scala

```diff
@@ -0,0 +1,35 @@
+package io.iohk.ethereum.db.dataSource
+
+import java.nio.ByteBuffer
+import java.util.concurrent.atomic.AtomicLong
+
+import io.iohk.iodb.{ByteArrayWrapper, LSMStore}
+
+class IodbDataSource(lSMStore: LSMStore) extends DataSource {
+
+  import IodbDataSource._
+
+  override def get(namespace: Byte, key: Key): Option[Value] =
+    lSMStore.get(ByteArrayWrapper((namespace +: key).toArray)).map(v => v.data.toIndexedSeq)
+
+  override def update(namespace: Byte, toRemove: Seq[Key], toUpdate: Seq[(Key, Value)]): DataSource = {
```

Comment:
I rechecked how `IODB` handles that problem, and although it throws an exception in that case, it only does so after some version updates and inserting the keys which have the correct length. I agree we should handle this but it will need some refactoring that's also necessary for `EC-60`. So what do you think on leaving it right now as it is, so as to not delay much more other PRs, and I can add the handling of this in my next PR for `EC-60` or maybe in another PR?